### ISSUE TITLE
add optimization subpackage in `functorch`

### ISSUE THREAD

**Issue by alibool** [pytorch/functorch#985]
Jax has its optimization subpackage in `jax.scipy.optimize` but `functorch` does not, [this repo](https://github.com/rfeinman/pytorch-minimize) may provide some help.

**Comment by richinex** [comment]
You're right alibool. Reuben Feinman's library Pytorch-minimize is awesome. Functorch should have 
something  equivalent to jsoptimize
